experimental/generation/generator/packages/library/src/mergeAssets.ts

```diff
@@ -294,148 +292,189 @@ async function mergeRootFile(schemaName: string, oldPath: string, oldFileList: s
-                    let j = i + 1
-                    while (j < newEntitySection.ListBody.length) {
-                        if (!newEntitySection.ListBody[j].match(':')) {
-                            temp.push(newEntitySection.ListBody[j])
-                            j++
-                            if (j === newEntitySection.ListBody.length) {
-                                enumValueMap.set(enumEntity, temp)
-                            }
-                        } else {
-                            enumValueMap.set(enumEntity, temp)
-                            i = j - 1
-                            break
-                        }
-                    }
-                }
-            }
+    let patternIndex = oldText.search(GeneratorPattern)
+    if (patternIndex !== -1) {
+        val = val + os.EOL + oldText.substring(patternIndex)
+    }
 
-            //parse old lu entity list and delete the enum entity which does not exist in new lu file
-            for (let i = 0; i < oldListEntitySection.ListBody.length; i++) {
-                // if the string has : (e.g., - multiGrainWheat :), parse it as an enum entity
-                if (oldListEntitySection.ListBody[i].match(':')) {
-                    const enumEntity = oldListEntitySection.ListBody[i].replace('-', '').replace(':', '').trim()
-                    enumSet.add(enumEntity)
-                    if (enumValueMap.has(enumEntity)) {
-                        resultStatements.push(oldListEntitySection.ListBody[i])
-                    }
-                    let j = i + 1
-                    while (j < oldListEntitySection.ListBody.length) {
-                        if (!oldListEntitySection.ListBody[j].match(':')) {
-                            const enumSyn = oldListEntitySection.ListBody[j].replace('-', '').trim()
-                            if (enumValueMap.has(enumEntity)) {
-                                resultStatements.push(oldListEntitySection.ListBody[j])
-                                keepEnumValue.add(enumSyn)
-                            } else {
-                                deletedEnumValue.add(enumSyn)
-                            }
-                            j++
-                        } else {
-                            i = j - 1
-                            break
-                        }
-                    }
-                }
-            }
+    await writeToFile(oldPath, mergedPath, `${schemaName}.${locale}.lu`, oldFileList, val, feedback)
+}
 
-            // add  new enum entity in the new  lu file 
-            for (const [key, values] of enumValueMap) {
-                if (!enumSet.has(key)) {
-                    resultStatements.push('    - ' + key + ' :')
-                    for (const newStatement of values) {
-                        resultStatements.push(newStatement)
-                    }
-                }
-            }
+const valuePattern = /(?<open>{@)(?<lable>[^=]+)=((?<value>[a-zA-Z]*)|($1))/g
```

Comment:
>lable [](start = 36, length = 5)

label

Reply:
I find that typescript does not support balancing group.
(?<open>{@)(?<label>[^=]+)=(?<value>.*)(?<-open>}) is illegal for typescript.

Reply:
I do not find the regex pattern for typescript which is similar to balancing group supported by .Net